health/checker.go

```diff
@@ -0,0 +1,125 @@
+package health
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+)
+
+const (
+	// The image tag is set to the value of the IMAGE_TAG environment variable,
+	// which is passed to the Docker image as a build argument at build time.
+	// It represent the semver version of PATH (eg. `v0.0.1`).
+	imageTagEnvVar = "IMAGE_TAG"
+	// If the image tag is not set by the Docker build process, the default value is "development".
+	defaultImageTag = "development"
+)
+
+// The status of the health check component.
+type healthCheckStatus string
+
+const (
+	// StatusReady indicates that all PATH components are ready
+	statusReady healthCheckStatus = "ready"
+	// StatusNotReady indicates that one or more PATH components
+	// are still initializing (e.g. warming up caches, etc)
+	statusNotReady healthCheckStatus = "not_ready"
+)
+
+type (
+	// The health.Checker struct is used to store all PATH components whose
+	// health needs to be checked to consider PATH ready to serve traffic.
+	Checker struct {
```

Comment:
Does this need to be publically exposed?

Reply:
Yes, it's use in other packages (`main`, `relayer` and the relayer `morse` and `shannon` packages).